lib/layouts/otp_verification/otp_enter_phone.dart

```diff
@@ -1,59 +1,126 @@
+    this.customSelectorConfig = const SelectorConfig(
+      selectorType: PhoneInputSelectorType.BOTTOM_SHEET,
+      useBottomSheetSafeArea: true,
+      leadingPadding: 20,
+      setSelectorButtonAsPrefixIcon: true,
+    ),
+    this.customAutoValidateMode = AutovalidateMode.disabled,
+    this.phoneNumber,
+    this.customInputDecoration = const InputDecoration(),
+    this.customSearchBoxInputDecoration = const InputDecoration(),
+    this.customInputBorder = const OutlineInputBorder(),
+    this.formatInput = false,
+    this.onSubmit,
+    this.onValidate,
+    this.onFieldSubmitted,
   });
 
+  @override
+  State<OTPEnterPhoneNumber> createState() => _OTPEnterPhoneNumberState();
+}
+
+class _OTPEnterPhoneNumberState extends State<OTPEnterPhoneNumber> {
   @override
   Widget build(BuildContext context) {
     return LayoutWrapper(
       child: Column(
         mainAxisAlignment: MainAxisAlignment.center,
         children: [
-          if (logo != null) ...[
-            logo!,
+          if (widget.logo != null) ...[
+            widget.logo!,
             const SizedBox(height: 40),
           ],
-          if (descriptionText.isNotEmpty)
+          if (widget.descriptionText.isNotEmpty)
             Padding(
               padding: const EdgeInsets.symmetric(horizontal: 16.0),
               child: Text(
-                descriptionText,
+                widget.descriptionText,
                 style: context.textStyle(TextScale.bodyMedium),
                 textAlign: TextAlign.center,
               ),
             ),
           const SizedBox(height: 40),
-          // TODO(roman.ovsepian) Replace with PhoneNumberInput
-          EWTextField(
-            hintText: phoneNumberHintText,
+          InternationalPhoneNumberInput(
+            onInputChanged: (PhoneNumber number) {
+              widget.onInputChanged?.call(number);
+            },
+            onInputValidated: (bool value) {
+              widget.onInputValidated?.call(value);
+            },
+            selectorConfig: widget.customSelectorConfig,
+            ignoreBlank: false,
+            autoValidateMode: widget.customAutoValidateMode,
+            selectorTextStyle: const TextStyle(color: Colors.black),
+            initialValue: widget.phoneNumber ?? PhoneNumber(isoCode: 'UA'),
+            textFieldController: widget.controller,
+            inputDecoration: widget.customInputDecoration,
+            searchBoxDecoration: widget.customSearchBoxInputDecoration,
+            formatInput: widget.formatInput,
+            keyboardType: const TextInputType.numberWithOptions(
+                signed: true, decimal: true),
+            inputBorder: widget.customInputBorder,
+            onSaved: (PhoneNumber number) {
+              widget.onSaved?.call(number);
+            },
+            onSubmit: () {
+              widget.onSubmit?.call();
+            },
+            validator: (String? value) {
+              return widget.onValidate?.call(value ?? '');
```

Comment:
try to simplify it like that: `validator: widget.onValidate`

Reply:
cant figure out how to do that. We get   final String? Function(String?)? validator,  in the phone number input package, and it seems  String value is mandatory